common/src/main/java/com/aliyun/rss/common/network/server/TransportRequestHandler.java

```diff
@@ -0,0 +1,290 @@
+      TransportClient reverseClient,
+      RpcHandler rpcHandler,
+      Long maxChunksBeingTransferred) {
+    this.channel = channel;
+    this.reverseClient = reverseClient;
+    this.rpcHandler = rpcHandler;
+    this.streamManager = rpcHandler.getStreamManager();
+    this.maxChunksBeingTransferred = maxChunksBeingTransferred;
+  }
+
+  @Override
+  public void exceptionCaught(Throwable cause) {
+    rpcHandler.exceptionCaught(cause, reverseClient);
+  }
+
+  @Override
+  public void channelActive() {
+    rpcHandler.channelActive(reverseClient);
+  }
+
+  @Override
+  public void channelInactive() {
+    if (streamManager != null) {
+      try {
+        streamManager.connectionTerminated(channel);
+      } catch (RuntimeException e) {
+        logger.error("StreamManager connectionTerminated() callback failed.", e);
+      }
+    }
+    rpcHandler.channelInactive(reverseClient);
+  }
+
+  @Override
+  public void handle(RequestMessage request) {
+    if (request instanceof ChunkFetchRequest) {
+      if (checkRegistered(request)) {
+        processFetchRequest((ChunkFetchRequest) request);
+      }
+    } else if (request instanceof RpcRequest) {
+      if (checkRegistered(request)) {
+        processRpcRequest((RpcRequest) request);
+      }
+    } else if (request instanceof OneWayMessage) {
+      if (checkRegistered(request)) {
+        processOneWayMessage((OneWayMessage) request);
+      }
+    } else if (request instanceof PushData) {
+      if (checkRegistered(request)) {
+        processPushData((PushData) request);
+      }
+    } else if (request instanceof PushMergedData) {
+      if (checkRegistered(request)) {
+        processPushMergedData((PushMergedData) request);
+      }
+    } else {
+      throw new IllegalArgumentException("Unknown request type: " + request);
+    }
+  }
+
+  private boolean checkRegistered(RequestMessage req) {
+    if (!rpcHandler.checkRegistered()) {
+      IOException e = new IOException("Worker Not Registered!");
+      if (req instanceof RpcRequest) {
+        respond(new RpcFailure(((RpcRequest)req).requestId, Throwables.getStackTraceAsString(e)));
+      } else if (req instanceof ChunkFetchRequest) {
+        respond(new ChunkFetchFailure(((ChunkFetchRequest)req).streamChunkId,
+            Throwables.getStackTraceAsString(e)));
+      } else if (req instanceof OneWayMessage) {
+        logger.warn("Ignore OneWayMessage since worker is not registered!");
+      }
+      return false;
+    }
+    return true;
+  }
+
+  private void processFetchRequest(final ChunkFetchRequest req) {
+    if (source != null) {
+      source.startTimer(NetWorkSource.FetchChunkTime(), req.toString());
+    }
+    if (logger.isTraceEnabled()) {
+      logger.trace("Received req from {} to fetch block {}", NettyUtils.getRemoteAddress(channel),
+        req.streamChunkId);
+    }
+    long chunksBeingTransferred = streamManager.chunksBeingTransferred();
+    if (chunksBeingTransferred >= maxChunksBeingTransferred) {
```

Comment:
I think we do not need to close the channel, instead we can just wait until the request can be fulfilled. BTW, why we need this check, what is the problem if we sever many reading request concurrently?